samples/snippets/src/test/java/com/example/dialogflow/CreateConversationProfileTest.java

```diff
@@ -0,0 +1,139 @@
+    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Before
+  public void setUp() {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    if (conversationProfileNameToDelete != null) {
+      deleteConversationProfile(conversationProfileNameToDelete);
+      conversationProfileNameToDelete = null;
+    }
+
+    System.setOut(null);
+  }
+
+  @Test
+  public void testCreateConversationProfileArticleFaq() throws IOException {
+    String conversationProfileDisplayName = UUID.randomUUID().toString();
+
+    // Create a conversation profile
+    String articleSuggestionKnowledgeBaseId = UUID.randomUUID().toString();
+    String faqKnowledgeBaseId = UUID.randomUUID().toString();
+    ConversationProfile createdConversationProfile =
+        ConversationProfileManagement.createConversationProfileArticleFaq(
+            PROJECT_ID,
+            conversationProfileDisplayName,
+            LOCATION,
+            Optional.of(articleSuggestionKnowledgeBaseId),
+            Optional.of(faqKnowledgeBaseId));
+    conversationProfileNameToDelete = createdConversationProfile.getName();
+    assertEquals(conversationProfileDisplayName, createdConversationProfile.getDisplayName());
+    List<SuggestionFeatureConfig> featureConfigsList =
+        createdConversationProfile
+            .getHumanAgentAssistantConfig()
+            .getHumanAgentSuggestionConfig()
+            .getFeatureConfigsList();
+    assertEquals(2, featureConfigsList.size());
+    assertTrue(
+        featureConfigsList.stream()
+            .filter(
+                featureConfig ->
+                    featureConfig.getSuggestionFeature().getType().equals(Type.ARTICLE_SUGGESTION))
+            .anyMatch(
+                featureConfig ->
+                    KnowledgeBaseName.parse(
+                            featureConfig
+                                .getQueryConfig()
+                                .getKnowledgeBaseQuerySource()
+                                .getKnowledgeBases(0))
+                        .getKnowledgeBase()
+                        .equals(articleSuggestionKnowledgeBaseId)));
+    assertTrue(
+        featureConfigsList.stream()
+            .filter(
+                featureConfig -> featureConfig.getSuggestionFeature().getType().equals(Type.FAQ))
+            .anyMatch(
+                featureConfig ->
+                    KnowledgeBaseName.parse(
+                            featureConfig
+                                .getQueryConfig()
+                                .getKnowledgeBaseQuerySource()
+                                .getKnowledgeBases(0))
+                        .getKnowledgeBase()
+                        .equals(faqKnowledgeBaseId)));
+
+    // Delete the conversation profile
+    deleteConversationProfile(conversationProfileNameToDelete);
+    conversationProfileNameToDelete = null;
```

Comment:
It looks like a failure will leave a lot of conversation Profile's around.  Perhaps you'd like to make things a bit more robust.

Reply:
It could be mitigated by checking `conversationProfileNameToDelete` during `tearDown`. Do you have other suggestions?

Reply:
We often look for >24h old resources in the `setUp()` method and delete them.  That way if we crash, they are eventually removed.